acceptance/tests/vault/vault_partitions_test.go

```diff
@@ -0,0 +1,250 @@
+	consulCluster.Create(t)
+
+	partitionServiceName := fmt.Sprintf("%s-consul-partition", consulReleaseName)
+	partitionSvcAddress := k8s.ServiceHost(t, cfg, serverClusterCtx, partitionServiceName)
+
+	k8sAuthMethodHost := k8s.KubernetesAPIServerHost(t, cfg, clientClusterCtx)
+
+	// Move Vault CA secret from primary to secondary so that we can mount it to pods in the
+	// secondary cluster.
+	logger.Logf(t, "retrieving Vault CA secret %s from the primary cluster and applying to the secondary", vaultCASecretName)
+	vaultCASecret, err := serverClusterCtx.KubernetesClient(t).CoreV1().Secrets(ns).Get(context.Background(), vaultCASecretName, metav1.GetOptions{})
+	vaultCASecret.ResourceVersion = ""
+	require.NoError(t, err)
+	_, err = clientClusterCtx.KubernetesClient(t).CoreV1().Secrets(ns).Create(context.Background(), vaultCASecret, metav1.CreateOptions{})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		clientClusterCtx.KubernetesClient(t).CoreV1().Secrets(ns).Delete(context.Background(), vaultCASecretName, metav1.DeleteOptions{})
+	})
+
+	// Create client cluster.
+	clientHelmValues := map[string]string{
+		"global.enabled": "false",
+
+		"global.adminPartitions.name": secondaryPartition,
+
+		"global.acls.bootstrapToken.secretName": "consul/data/secret/partition",
+		"global.acls.bootstrapToken.secretKey":  "token",
+
+		"global.secretsBackend.vault.agentAnnotations":    fmt.Sprintf("vault.hashicorp.com/tls-server-name: %s-vault", vaultReleaseName),
+		"global.secretsBackend.vault.adminPartitionsRole": "partition-init",
+
+		"externalServers.enabled":           "true",
+		"externalServers.hosts[0]":          partitionSvcAddress,
+		"externalServers.tlsServerName":     "server.dc1.consul",
+		"externalServers.k8sAuthMethodHost": k8sAuthMethodHost,
+
+		"client.enabled":           "true",
+		"client.exposeGossipPorts": "true",
+		"client.join[0]":           partitionSvcAddress,
+	}
+
+	if cfg.UseKind {
+		clientHelmValues["externalServers.httpsPort"] = "30000"
+		clientHelmValues["meshGateway.service.type"] = "NodePort"
+		clientHelmValues["meshGateway.service.nodePort"] = "30100"
+	}
+
+	helpers.MergeMaps(clientHelmValues, commonHelmValues)
+
+	// Install the consul cluster without servers in the client cluster kubernetes context.
+	clientConsulCluster := consul.NewHelmCluster(t, clientHelmValues, clientClusterCtx, cfg, consulReleaseName)
+	clientConsulCluster.Create(t)
+
+	// Ensure consul clients are created.
+	agentPodList, err := clientClusterCtx.KubernetesClient(t).CoreV1().Pods(clientClusterCtx.KubectlOptions(t).Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=consul,component=client"})
+	require.NoError(t, err)
+	require.NotEmpty(t, agentPodList.Items)
+
+	output, err := k8s.RunKubectlAndGetOutputE(t, clientClusterCtx.KubectlOptions(t), "logs", agentPodList.Items[0].Name, "consul", "-n", clientClusterCtx.KubectlOptions(t).Namespace)
+	require.NoError(t, err)
+	require.Contains(t, output, "Partition: 'secondary'")
```

Comment:
no need to test cross-partition communication here I think because we mostly care that partition bootstrapping worked. But let me know if you think otherwise

Reply:
👍 I added discussing this to our next agenda because I am curious about cutting my teeth on writing a new acceptance test and for this, I am curious just to talk through how we might do this here...but I don't think it's necessary at the moment.

Reply:
Now that the infrastructure is in place, it should not be too hard to augment this test to also test x-partition @jmurret. It shouldn't be dissimilar for the existing partitions test.

As far as the rationale for not testing it, SGTM. This is testing the Vault integration and not the partitions test.